sklearn/utils/stats.py

```diff
@@ -1,18 +1,44 @@
 import numpy as np
 
 from .extmath import stable_cumsum
+from sklearn.utils.fixes import _take_along_axis
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
     """
     Compute the weighted ``percentile`` of ``array`` with ``sample_weight``.
+    If ``array`` is 2D, compute weighted ``percentile`` along axis=0.
     """
-    sorted_idx = np.argsort(array)
+    if sample_weight is None:
+        sample_weight = np.ones_like(array)
+    array = np.squeeze(array)
+    sample_weight = np.squeeze(sample_weight)
+
+    n_dim = array.ndim
+    if n_dim == 0:
+        return array[()]
+    sorted_idx = np.argsort(array, axis=0)
+    sorted_weights = _take_along_axis(sample_weight, sorted_idx, axis=0)
+    sorted_array = _take_along_axis(array, sorted_idx, axis=0)
```

Comment:
why is this needed rather than using the `sorted_idx` directly in looking up the `percentile_idx`?

Reply:
I didn't know how to make this work for 2D array? Can I do a nested `take_along_axis`?

Reply:
Fancy indexing!